src/univariate/continuous/normal.jl

```diff
@@ -1,12 +1,18 @@
-immutable Normal <: ContinuousUnivariateDistribution
-    μ::Float64
-    σ::Float64
+immutable Normal{T<:Number} <: ContinuousUnivariateDistribution
+    μ::T
+    σ::T
 
-    Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); new(μ, σ))
-    Normal(μ::Real) = Normal(μ, 1.0)
-    Normal() = Normal(0.0, 1.0)
+    Normal(μ::T, σ::T) = new(μ, σ)
 end
 
+Normal{T<:Number}(μ::T, σ::T) = Normal{T}(μ, σ)
+Normal(μ::Number, σ::Number) = Normal(promote(μ, σ)...)
+Normal(μ::Number) = Normal(μ, one(μ))
+#Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); Normal(μ, σ))
+Normal(μ::Real) = Normal(μ, one(μ))
+Normal() = Normal(0.0, 1.0)
+Normal(μ::Complex, σ::Complex) = error("Normal with complex mean and variance not defined")
```

Comment:
This was very interesting too... one more link (if it opens, page 21, section 1.4   https://books.google.co.uk/books?id=eo_u18h2REIC&pg=PA29&lpg=PA29&dq=hyperreal+computing&source=bl&ots=sDDyZK9cyE&sig=PAYurkn77hJdpe6VjPx1O7x_87w&hl=en&sa=X&ved=0CCIQ6AEwAmoVChMIn_KB5MLZyAIVBvEUCh3q9g6v#v=onepage&q=hyperreal%20computing&f=false). I will read this nice intro section of the book tomorrow... About computing, I presume the easiest thing to do is to let functions take Number arguments for AD purposes probably.